app/src/main/java/com/nextcloud/talk/webrtc/MagicPeerConnectionWrapper.java

```diff
@@ -145,15 +154,17 @@ public void removePeerConnection() {
         if (magicDataChannel != null) {
             magicDataChannel.dispose();
             magicDataChannel = null;
+            Log.d(TAG, "Disposed DataChannel");
+        } else {
+            Log.d(TAG, "DataChannel is null.");
         }
 
         if (peerConnection != null) {
-            if (localMediaStream != null) {
-                peerConnection.removeStream(localMediaStream);
-            }
-
-            peerConnection.close();
+            peerConnection.dispose();
```

Comment:
Because of the by @mahibi reported crash (see https://github.com/nextcloud/talk-android/pull/1774#issuecomment-1044353103) I removed the `dispose()`.

Nevertheless, I have added a brief explanation for the current change : https://github.com/nextcloud/talk-android/pull/1774/commits/3be2c3ea57bef155d1bd840237b65bb1fb94f6c9

Reply:
Please note that I am totally clueless regarding the handling of streams and their lifetime in the Android library. But I wonder why the local stream was being removed from the connection before closing it in the original code; I would not expect that to be needed, unless the stream was kept alive by being referenced by a closed connection (which would be strange, but possible nevertheless).

Similarly, I wonder if explicitly disposing the local media stream is needed, or if just nullifying it (along with closing the connections using it) would be enough to automatically stop the stream. Explicitly stopping and freeing its resources (which I guess is what `dispose` does, but I do not know :-) ) seems cleaner, but as it is being done before closing the connections using the stream... I do not know if that could cause any issue depending on how fast the stream and the connection are internally disposed and closed (although I would expect the library to shield against something like that). Maybe it could be verified if it is safe by disposing the stream and, five seconds later or something like that, close the connections.

Reply:
I move the `localStream.dispose()` behind the closing of the `peerConnection`. I think your're right. That order makes more sense.

Without calling active `dispose` it will never be called. I'm not sure what on the native site happens. But during the `dispose` many JNI methods will be called and giving free resources. An it is checked that the resource is not already disposed. 

At this point I will go with it and improve the current WebRTC situation in later iterations.